21.0.0.alpha5 endpoints compile error

### Description of the bug / feature
Starting with 21.0.0.alpha5, the TS generated for certain endpoints failed to compile with production profile.
For example I've got a java class called MessageEndpoint with a method with this signature

```
public void send(String text, int chatPartnerId)
```

the generated typescript is then

```
function _send(
 text?: string,
 chatPartnerId: number
): Promise<void> {
 return client.call('MessageEndpoint', 'send', {text, chatPartnerId});
}
```

however this does not compile: 

```
[ERROR] TS1016: A required parameter cannot follow an optional parameter.
[ERROR]     19 | function _send(
[ERROR]     20 |  text?: string,
[ERROR]   > 21 |  chatPartnerId: number
[ERROR]        |  ^^^^^^^^^^^^^
[ERROR]     22 | ): Promise<void> {
[ERROR]     23 |  return client.call('MessageEndpoint', 'send', {text, chatPartnerId});
[ERROR]     24 | }

````



### Minimal reproducible example
create a new java endpoint with the given signature, try to compile it with the production profile 

```
mvn -P production -DskipTests=true clean dependency:list install 
```
### Expected behavior
typescript code should compile

### Actual behavior
see above
### Versions:
Vaadin 21.0.0.alpha5, maven 3, java = corretto-11.0.11

--- a/fusion-endpoint/src/main/java/com/vaadin/flow/server/connect/generator/OpenApiObjectGenerator.java
+++ b/fusion-endpoint/src/main/java/com/vaadin/flow/server/connect/generator/OpenApiObjectGenerator.java
@@ -709,11 +709,11 @@ public class OpenApiObjectGenerator {
                         paramsDescription.remove(parameter.getNameAsString()));
             }
             requestSchema.addProperties(name, paramSchema);
-            if (GeneratorUtils.isNotTrue(paramSchema.getNullable())
-                    || isRequired(parameter)) {
-                requestSchema.addRequiredItem(name);
+            requestSchema.addRequiredItem(name);
+            if (GeneratorUtils.isTrue(paramSchema.getNullable())
+                    && isRequired(parameter)) {
+                paramSchema.setNullable(null);
             }
-            paramSchema.setNullable(null);
         });
         if (!paramsDescription.isEmpty()) {
             requestSchema.addExtension(

--- a/fusion-endpoint/src/test/java/com/vaadin/flow/server/connect/generator/endpoints/AbstractEndpointGenerationTest.java
+++ b/fusion-endpoint/src/test/java/com/vaadin/flow/server/connect/generator/endpoints/AbstractEndpointGenerationTest.java
@@ -351,14 +351,7 @@ public abstract class AbstractEndpointGenerationTest
             assertSchema(stringSchemaEntry.getValue(), parameterTypes[index],
                     parameterTypeArguments.get(index));
             List requiredList = requestSchema.getRequired();
-            if (isRequired(parameters[index])) {
-                assertTrue(requiredList.contains(stringSchemaEntry.getKey()));
-            } else {
-                boolean notRequired = requiredList == null
-                        || !requiredList.contains(stringSchemaEntry.getKey());
-                assertTrue("Request parameter without @Nonnull annotation"
-                        + " should not be required", notRequired);
-            }
+            assertTrue(requiredList.contains(stringSchemaEntry.getKey()));
             index++;
         }
     }

--- a/fusion-endpoint/src/test/java/com/vaadin/flow/server/connect/generator/endpoints/model/ComplexTypeParamsEndpoint.java
+++ b/fusion-endpoint/src/test/java/com/vaadin/flow/server/connect/generator/endpoints/model/ComplexTypeParamsEndpoint.java
@@ -25,4 +25,8 @@ public class ComplexTypeParamsEndpoint {
     public void getComplexTypeParams(List<ModelEndpoint.Account> accounts,
             Map<String, ModelEndpoint.Group> groups) {
     }
+
+    public void send(String text, int chatPartnerId) {
+        
+    }
 }

--- a/fusion-endpoint/src/test/java/com/vaadin/flow/server/connect/generator/endpoints/model/ComplexTypeParamsEndpoint.java
+++ b/fusion-endpoint/src/test/java/com/vaadin/flow/server/connect/generator/endpoints/model/ComplexTypeParamsEndpoint.java
@@ -27,6 +27,6 @@ public class ComplexTypeParamsEndpoint {
     }
 
     public void send(String text, int chatPartnerId) {
-        
+
     }
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/collectionendpoint/expected-CollectionEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/collectionendpoint/expected-CollectionEndpoint.ts
@@ -15,7 +15,7 @@ import Collection from './com/vaadin/flow/server/connect/generator/endpoints/col
  * Return a collection
  */
 function _getCollectionByAuthor(
-    name?: string
+    name: string | undefined
 ): Promise<Collection | undefined> {
   return client.call('CollectionEndpoint', 'getCollectionByAuthor', {name});
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/datetime/expected-DateTimeEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/datetime/expected-DateTimeEndpoint.ts
@@ -8,49 +8,49 @@
 import client from './connect-client.default';
 
 function _echoDate(
-    date?: string
+    date: string | undefined
 ): Promise<string | undefined> {
   return client.call('DateTimeEndpoint', 'echoDate', {date});
 }
 export {_echoDate as echoDate};
 
 function _echoInstant(
-    instant?: string
+    instant: string | undefined
 ): Promise<string | undefined> {
   return client.call('DateTimeEndpoint', 'echoInstant', {instant});
 }
 export {_echoInstant as echoInstant};
 
 function _echoListLocalDateTime(
-    localDateTimeList?: Array<string | undefined>
+    localDateTimeList: Array<string | undefined> | undefined
 ): Promise<Array<string | undefined> | undefined> {
   return client.call('DateTimeEndpoint', 'echoListLocalDateTime', {localDateTimeList});
 }
 export {_echoListLocalDateTime as echoListLocalDateTime};
 
 function _echoLocalDate(
-    localDate?: string
+    localDate: string | undefined
 ): Promise<string | undefined> {
   return client.call('DateTimeEndpoint', 'echoLocalDate', {localDate});
 }
 export {_echoLocalDate as echoLocalDate};
 
 function _echoLocalDateTime(
-    localDateTime?: string
+    localDateTime: string | undefined
 ): Promise<string | undefined> {
   return client.call('DateTimeEndpoint', 'echoLocalDateTime', {localDateTime});
 }
 export {_echoLocalDateTime as echoLocalDateTime};
 
 function _echoLocalTime(
-    localTime?: string
+    localTime: string | undefined
 ): Promise<string | undefined> {
   return client.call('DateTimeEndpoint', 'echoLocalTime', {localTime});
 }
 export {_echoLocalTime as echoLocalTime};
 
 function _echoMapInstant(
-    mapInstant?: Record<string, string | undefined>
+    mapInstant: Record<string, string | undefined> | undefined
 ): Promise<Record<string, string | undefined> | undefined> {
   return client.call('DateTimeEndpoint', 'echoMapInstant', {mapInstant});
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/enumtype/expected-EnumEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/enumtype/expected-EnumEndpoint.ts
@@ -9,14 +9,14 @@ import client from './connect-client.default';
 import MyEnum from './com/vaadin/flow/server/connect/generator/endpoints/enumtype/EnumEndpoint/MyEnum';
 
 function _echoEnum(
-    value?: MyEnum
+    value: MyEnum | undefined
 ): Promise<MyEnum | undefined> {
   return client.call('EnumEndpoint', 'echoEnum', {value});
 }
 export {_echoEnum as echoEnum};
 
 function _echoListEnum(
-    enumList?: Array<MyEnum | undefined>
+    enumList: Array<MyEnum | undefined> | undefined
 ): Promise<Array<MyEnum | undefined> | undefined> {
   return client.call('EnumEndpoint', 'echoListEnum', {enumList});
 }
@@ -28,7 +28,7 @@ function _getEnum(): Promise<MyEnum | undefined> {
 export {_getEnum as getEnum};
 
 function _setEnum(
-    value?: MyEnum
+    value: MyEnum | undefined
 ): Promise<void> {
   return client.call('EnumEndpoint', 'setEnum', {value});
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/inheritedmodel/expected-InheritedModelEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/inheritedmodel/expected-InheritedModelEndpoint.ts
@@ -10,7 +10,7 @@ import ChildModel from './com/vaadin/flow/server/connect/generator/endpoints/inh
 import ParentModel from './com/vaadin/flow/server/connect/generator/endpoints/inheritedmodel/InheritedModelEndpoint/ParentModel';
 
 function _getParentModel(
-    child?: ChildModel
+    child: ChildModel | undefined
 ): Promise<ParentModel | undefined> {
   return client.call('InheritedModelEndpoint', 'getParentModel', {child});
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/json/expected-JsonTestEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/json/expected-JsonTestEndpoint.ts
@@ -29,7 +29,7 @@ export {_countUser as countUser};
  * Return current time as an Instant
  */
 function _fullFQNMethod(
-    input?: number
+    input: number | undefined
 ): Promise<string | undefined> {
   return client.call('JsonTestEndpoint', 'fullFQNMethod', {input});
 }
@@ -62,7 +62,7 @@ export {_getAllUsers as getAllUsers};
  * Return array of int
  */
 function _getArrayInt(
-    input?: Array<string | undefined>
+    input: Array<string | undefined> | undefined
 ): Promise<Array<number> | undefined> {
   return client.call('JsonTestEndpoint', 'getArrayInt', {input});
 }
@@ -75,7 +75,7 @@ export {_getArrayInt as getArrayInt};
  * Return boolean value
  */
 function _getBooleanValue(
-    input?: Record<string, User | undefined>
+    input: Record<string, User | undefined> | undefined
 ): Promise<boolean> {
   return client.call('JsonTestEndpoint', 'getBooleanValue', {input});
 }
@@ -89,7 +89,7 @@ export {_getBooleanValue as getBooleanValue};
  * Return boolean value
  */
 function _getTwoParameters(
-    input?: string,
+    input: string | undefined,
     secondInput: number
 ): Promise<boolean> {
   return client.call('JsonTestEndpoint', 'getTwoParameters', {input, secondInput});
@@ -110,22 +110,22 @@ function _getUserById(
 export {_getUserById as getUserById};
 
 function _inputBeanTypeDependency(
-    input?: Version
+    input: Version | undefined
 ): Promise<void> {
   return client.call('JsonTestEndpoint', 'inputBeanTypeDependency', {input});
 }
 export {_inputBeanTypeDependency as inputBeanTypeDependency};
 
 function _inputBeanTypeLocal(
-    input?: Status
+    input: Status | undefined
 ): Promise<void> {
   return client.call('JsonTestEndpoint', 'inputBeanTypeLocal', {input});
 }
 export {_inputBeanTypeLocal as inputBeanTypeLocal};
 
 function _optionalParameter(
-    parameter?: Array<string | undefined>,
-    requiredParameter?: string
+    parameter: Array<string | undefined> | undefined,
+    requiredParameter: string | undefined
 ): Promise<void> {
   return client.call('JsonTestEndpoint', 'optionalParameter', {parameter, requiredParameter});
 }
@@ -150,7 +150,7 @@ export {_reservedWordInParameter as reservedWordInParameter};
  *
  */
 function _updateUser(
-    user?: User
+    user: User | undefined
 ): Promise<void> {
   return client.call('JsonTestEndpoint', 'updateUser', {user});
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/json/expected-openapi-custom-application-properties.json
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/json/expected-openapi-custom-application-properties.json
@@ -561,7 +561,7 @@
       "com.vaadin.flow.server.connect.generator.endpoints.json.JsonTestEndpoint.Permissions" : {
         "type" : "string",
         "enum" : [ "ADMIN", "USER", "GUEST" ],
-        "x-vaadin-file-path" : "D:\\Work\\flow\\fusion-endpoint\\src\\test\\java\\com\\vaadin\\flow\\server\\connect\\generator\\endpoints\\json\\JsonTestEndpoint.java"
+        "x-vaadin-file-path" : "com/vaadin/flow/server/connect/generator/endpoints/json/JsonTestEndpoint.java"
       },
       "com.vaadin.flow.server.connect.generator.endpoints.json.JsonTestEndpoint.Role" : {
         "type" : "object",

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/json/expected-openapi-custom-application-properties.json
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/json/expected-openapi-custom-application-properties.json
@@ -50,11 +50,13 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "input" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
                     "type" : "number",
-                    "description" : "input parameter"
+                    "description" : "input parameter",
+                    "nullable" : true
                   }
                 }
               }
@@ -150,11 +152,13 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "input" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
                     "type" : "array",
                     "description" : "input string array",
+                    "nullable" : true,
                     "items" : {
                       "type" : "string",
                       "nullable" : true
@@ -196,6 +200,7 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "input" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
@@ -206,7 +211,8 @@
                         "$ref" : "#/components/schemas/com.vaadin.flow.server.connect.generator.endpoints.json.JsonTestEndpoint.User"
                       } ]
                     },
-                    "description" : "input map"
+                    "description" : "input map",
+                    "nullable" : true
                   }
                 }
               }
@@ -240,12 +246,13 @@
           "content" : {
             "application/json" : {
               "schema" : {
-                "required" : [ "secondInput" ],
+                "required" : [ "input", "secondInput" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
                     "type" : "string",
-                    "description" : "first input description"
+                    "description" : "first input description",
+                    "nullable" : true
                   },
                   "secondInput" : {
                     "type" : "number",
@@ -324,9 +331,11 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "input" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
+                    "nullable" : true,
                     "allOf" : [ {
                       "$ref" : "#/components/schemas/com.fasterxml.jackson.core.Version"
                     } ]
@@ -354,9 +363,11 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "input" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
+                    "nullable" : true,
                     "allOf" : [ {
                       "$ref" : "#/components/schemas/com.vaadin.flow.server.connect.generator.endpoints.json.JsonTestEndpoint.Status"
                     } ]
@@ -384,17 +395,20 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "parameter", "requiredParameter" ],
                 "type" : "object",
                 "properties" : {
                   "parameter" : {
                     "type" : "array",
+                    "nullable" : true,
                     "items" : {
                       "type" : "string",
                       "nullable" : true
                     }
                   },
                   "requiredParameter" : {
-                    "type" : "string"
+                    "type" : "string",
+                    "nullable" : true
                   }
                 }
               }
@@ -474,10 +488,12 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "user" ],
                 "type" : "object",
                 "properties" : {
                   "user" : {
                     "description" : "User to be updated",
+                    "nullable" : true,
                     "allOf" : [ {
                       "$ref" : "#/components/schemas/com.vaadin.flow.server.connect.generator.endpoints.json.JsonTestEndpoint.User"
                     } ]
@@ -545,7 +561,7 @@
       "com.vaadin.flow.server.connect.generator.endpoints.json.JsonTestEndpoint.Permissions" : {
         "type" : "string",
         "enum" : [ "ADMIN", "USER", "GUEST" ],
-        "x-vaadin-file-path" : "com/vaadin/flow/server/connect/generator/endpoints/json/JsonTestEndpoint.java"
+        "x-vaadin-file-path" : "D:\\Work\\flow\\fusion-endpoint\\src\\test\\java\\com\\vaadin\\flow\\server\\connect\\generator\\endpoints\\json\\JsonTestEndpoint.java"
       },
       "com.vaadin.flow.server.connect.generator.endpoints.json.JsonTestEndpoint.Role" : {
         "type" : "object",

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/json/expected-openapi.json
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/json/expected-openapi.json
@@ -50,11 +50,13 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "input" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
                     "type" : "number",
-                    "description" : "input parameter"
+                    "description" : "input parameter",
+                    "nullable" : true
                   }
                 }
               }
@@ -150,11 +152,13 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "input" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
                     "type" : "array",
                     "description" : "input string array",
+                    "nullable" : true,
                     "items" : {
                       "type" : "string",
                       "nullable" : true
@@ -196,6 +200,7 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "input" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
@@ -206,7 +211,8 @@
                         "$ref" : "#/components/schemas/com.vaadin.flow.server.connect.generator.endpoints.json.JsonTestEndpoint.User"
                       } ]
                     },
-                    "description" : "input map"
+                    "description" : "input map",
+                    "nullable" : true
                   }
                 }
               }
@@ -240,12 +246,13 @@
           "content" : {
             "application/json" : {
               "schema" : {
-                "required" : [ "secondInput" ],
+                "required" : [ "input", "secondInput" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
                     "type" : "string",
-                    "description" : "first input description"
+                    "description" : "first input description",
+                    "nullable" : true
                   },
                   "secondInput" : {
                     "type" : "number",
@@ -324,9 +331,11 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "input" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
+                    "nullable" : true,
                     "allOf" : [ {
                       "$ref" : "#/components/schemas/com.fasterxml.jackson.core.Version"
                     } ]
@@ -354,9 +363,11 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "input" ],
                 "type" : "object",
                 "properties" : {
                   "input" : {
+                    "nullable" : true,
                     "allOf" : [ {
                       "$ref" : "#/components/schemas/com.vaadin.flow.server.connect.generator.endpoints.json.JsonTestEndpoint.Status"
                     } ]
@@ -384,17 +395,20 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "parameter", "requiredParameter" ],
                 "type" : "object",
                 "properties" : {
                   "parameter" : {
                     "type" : "array",
+                    "nullable" : true,
                     "items" : {
                       "type" : "string",
                       "nullable" : true
                     }
                   },
                   "requiredParameter" : {
-                    "type" : "string"
+                    "type" : "string",
+                    "nullable" : true
                   }
                 }
               }
@@ -474,10 +488,12 @@
           "content" : {
             "application/json" : {
               "schema" : {
+                "required" : [ "user" ],
                 "type" : "object",
                 "properties" : {
                   "user" : {
                     "description" : "User to be updated",
+                    "nullable" : true,
                     "allOf" : [ {
                       "$ref" : "#/components/schemas/com.vaadin.flow.server.connect.generator.endpoints.json.JsonTestEndpoint.User"
                     } ]

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/model/expected-ComplexTypeParamsEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/model/expected-ComplexTypeParamsEndpoint.ts
@@ -10,8 +10,8 @@ import Account from './com/vaadin/flow/server/connect/generator/endpoints/model/
 import Group from './com/vaadin/flow/server/connect/generator/endpoints/model/ModelEndpoint/Group';
 
 function _getComplexTypeParams(
-    accounts?: Array<Account | undefined>,
-    groups?: Record<string, Group | undefined>
+    accounts: Array<Account | undefined> | undefined,
+    groups: Record<string, Group | undefined> | undefined
 ): Promise<void> {
   return client.call('ComplexTypeParamsEndpoint', 'getComplexTypeParams', {accounts, groups});
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/model/expected-ComplexTypeParamsEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/model/expected-ComplexTypeParamsEndpoint.ts
@@ -17,6 +17,15 @@ function _getComplexTypeParams(
 }
 export {_getComplexTypeParams as getComplexTypeParams};
 
+function _send(
+  text: string | undefined,
+  chatPartnerId: number
+ ): Promise<void> {
+  return client.call('ComplexTypeParamsEndpoint', 'send', {text, chatPartnerId});
+ }
+export {_send as send};
+
 export const ComplexTypeParamsEndpoint = Object.freeze({
   getComplexTypeParams: _getComplexTypeParams,
+  send: _send,
 });

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/model/expected-ModelEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/model/expected-ModelEndpoint.ts
@@ -11,7 +11,7 @@ import Group from './com/vaadin/flow/server/connect/generator/endpoints/model/Mo
 import ModelFromDifferentPackage from './com/vaadin/flow/server/connect/generator/endpoints/model/subpackage/ModelFromDifferentPackage';
 
 function _getAccountByGroups(
-    groups?: Array<Group | undefined>
+    groups: Array<Group | undefined> | undefined
 ): Promise<Account | undefined> {
   return client.call('ModelEndpoint', 'getAccountByGroups', {groups});
 }
@@ -24,7 +24,7 @@ export {_getAccountByGroups as getAccountByGroups};
  * Return the account with given userName
  */
 function _getAccountByUserName(
-    userName?: string
+    userName: string | undefined
 ): Promise<Account | undefined> {
   return client.call('ModelEndpoint', 'getAccountByUserName', {userName});
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/modelpackage/expected-ModelPackageEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/modelpackage/expected-ModelPackageEndpoint.ts
@@ -25,7 +25,7 @@ export {_getListOfUserName as getListOfUserName};
  * Return a collection
  */
 function _getSameModelPackage(
-    name?: string
+    name: string | undefined
 ): Promise<Account | undefined> {
   return client.call('ModelPackageEndpoint', 'getSameModelPackage', {name});
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/modelpackage/expected-SubModelPackageEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/modelpackage/expected-SubModelPackageEndpoint.ts
@@ -9,7 +9,7 @@ import client from './connect-client.default';
 import Account from './com/vaadin/flow/server/connect/generator/endpoints/modelpackage/subpackage/Account';
 
 function _getSubAccountPackage(
-    name?: string
+    name: string | undefined
 ): Promise<Account | undefined> {
   return client.call('SubModelPackageEndpoint', 'getSubAccountPackage', {name});
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/nonnullable/expected-NonNullableEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/nonnullable/expected-NonNullableEndpoint.ts
@@ -33,7 +33,7 @@ function _getNotNullReturnType(): Promise<ReturnType | undefined> {
 export {_getNotNullReturnType as getNotNullReturnType};
 
 function _getNullableString(
-  input?: string
+  input: string | undefined
 ): Promise<string> {
   return client.call('NonNullableEndpoint', 'getNullableString', {input});
 }
@@ -41,7 +41,7 @@ function _getNullableString(
 export {_getNullableString as getNullableString};
 
 function _sendParameterType(
-  parameterType?: ParameterType
+  parameterType: ParameterType | undefined
 ): Promise<void> {
   return client.call('NonNullableEndpoint', 'sendParameterType', {parameterType});
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/sameclassname/expected-SameClassNameEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/sameclassname/expected-SameClassNameEndpoint.ts
@@ -10,21 +10,21 @@ import SameClassNameModel from './com/vaadin/flow/server/connect/generator/endpo
 import SubpackageSameClassNameModel from './com/vaadin/flow/server/connect/generator/endpoints/sameclassname/subpackage/SameClassNameModel';
 
 function _getMyClass(
-    sameClassNameModel?: Array<SubpackageSameClassNameModel | undefined>
+    sameClassNameModel: Array<SubpackageSameClassNameModel | undefined> | undefined
 ): Promise<SameClassNameModel | undefined> {
   return client.call('SameClassNameEndpoint', 'getMyClass', {sameClassNameModel});
 }
 export {_getMyClass as getMyClass};
 
 function _getSubpackageModelList(
-    sameClassNameModel?: Record<string, SubpackageSameClassNameModel | undefined>
+    sameClassNameModel: Record<string, SubpackageSameClassNameModel | undefined> | undefined
 ): Promise<Array<SubpackageSameClassNameModel | undefined> | undefined> {
   return client.call('SameClassNameEndpoint', 'getSubpackageModelList', {sameClassNameModel});
 }
 export {_getSubpackageModelList as getSubpackageModelList};
 
 function _getSubpackageModelMap(
-    sameClassNameModel?: Record<string, SameClassNameModel | undefined>
+    sameClassNameModel: Record<string, SameClassNameModel | undefined> | undefined
 ): Promise<Record<string, SubpackageSameClassNameModel | undefined> | undefined> {
   return client.call('SameClassNameEndpoint', 'getSubpackageModelMap', {sameClassNameModel});
 }
@@ -36,7 +36,7 @@ function _getSubpackageModel(): Promise<SubpackageSameClassNameModel | undefined
 export {_getSubpackageModel as getSubpackageModel};
 
 function _setSubpackageModel(
-    model?: SubpackageSameClassNameModel
+    model: SubpackageSameClassNameModel | undefined
 ): Promise<void> {
   return client.call('SameClassNameEndpoint', 'setSubpackageModel', {model});
 }

--- a/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/superclassmethods/expected-PersonEndpoint.ts
+++ b/fusion-endpoint/src/test/resources/com/vaadin/flow/server/connect/generator/endpoints/superclassmethods/expected-PersonEndpoint.ts
@@ -9,14 +9,14 @@ import client from './connect-client.default';
 import Person from './com/vaadin/flow/server/connect/generator/endpoints/superclassmethods/PersonEndpoint/Person';
 
 function _delete(
-    id?: number
+    id: number | undefined
 ): Promise<void> {
   return client.call('PersonEndpoint', 'delete', {id});
 }
 export {_delete as delete};
 
 function _get(
-    id?: number
+    id: number | undefined
 ): Promise<Person | undefined> {
   return client.call('PersonEndpoint', 'get', {id});
 }
@@ -36,7 +36,7 @@ function _size(): Promise<number> {
 export {_size as size};
 
 function _update(
-    entity?: Person
+    entity: Person | undefined
 ): Promise<Person | undefined> {
   return client.call('PersonEndpoint', 'update', {entity});
 }

